### DISCUSSION TITLE
Choose from device descriptors

### DISCUSSION THREAD

**Discussion by Nic7C5** [NicoHood/HID#313] [Ideas]
First of all you did a great job with this library.

I wanted to use a Sparkfun 32u4 board to translate rotary encoder increments to mouse/finger movements on a radial menu in an Android app. I managed to successfully test it on Windows using AbsoluteMouse but the Arduino isn't recognized when plugging it into any of my Android devices (that work with my Logitech USB Mouse). Is that due to incompatible device descriptors? I've seen the hex data in SingleAbsoluteMouse.cpp but I have no idea where to get data / full descriptors to test.

**Comment by NicoHood** [comment]
I am not sure if android supports absolute mouse devices at all.

@mdevaev do you have any hint?

**Reply by mdevaev** [reply]
I didn't have the opportunity to test Android, but I know for sure that the iPhone does not support an absolute mouse, it's probably the same with Android.

**Reply by mdevaev** [reply]
Oh, I have an idea. You can try adding the report_id to the descriptor. Of course, it will no longer be a single mouse (lol), but this hack works for iPhone, maybe the Android driver is the same freaky.
